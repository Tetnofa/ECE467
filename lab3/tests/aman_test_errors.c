// Test semantic errors (this should fail compilation)
int arr[3];  // Move array declaration to global scope

void errorTest() {
    int x;
    bool y;
    
    // Type mismatch
    x = true;  // Error
    
    // Undefined variable
    z = 5;     // Error
    
    // Invalid operation
    y = x + y; // Error
    
    // Array out of bounds
    arr[5] = 1; // Error
} 