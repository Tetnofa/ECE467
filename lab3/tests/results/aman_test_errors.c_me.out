sema: 9:4 : type mismatch 
sema: 12:4 : use of undefined identifier z
sema: 12:4 : type mismatch 
sema: 15:8 : type mismatch 
sema: 15:4 : type mismatch 
